packages/common/src/types/index.ts

```diff
@@ -57,6 +57,27 @@ export type Directory = {
   sourceId: string;
 };
 
+export type Template = {
+  name: TemplateType;
+  niceName: string;
+  shortid: string;
+  url: string;
+  main: boolean;
+  color: string;
+  backgroundColor: () => string | undefined;
+
+  popular: boolean;
+  showOnHomePage: boolean;
+  distDir: string;
+  netlify: boolean;
+  isTypescript: boolean;
+  externalResourcesEnabled: boolean;
+  showCube: boolean;
+  isServer: boolean;
+  mainFile: undefined | string[];
```

Comment:
@SaraVieira This can just be

```suggestion
  mainFile?: string[];
```